docs(guides): more context on how to configure the i18next plugin (#109)
sapphiredev/website: docs/Guide/plugins/i18next/getting-started.mdx

</TabItem>
</Tabs>

## Get the language set per server

> To obtain the language that each server has configured, as well as the custom prefixes for each server, for that we will
> have now an option called `i18n` in the Sapphire client options, in which we will do the following.

Reviewer: ```suggestion
Once you start to use i18next to offer your bot in different languages you are likely to also want each server to be able to configure their own language through some kind of configuration command while you store this data in a database. While this page will not cover those aspects, what we will cover here is how you can inform the i18next plugin about the language that should be used in the current context.

In order to achieve this, you will need to provide the options to the plugin. This is done through the `i18next` option on the [`ClientOptions`][saphclientoptions], specifically the [`fetchLanguage`] property on that object. This method receives 1 parameter, which is an object that has `message`, `guild`, and `user`, we call it the [`InternationalizationContext`]. Using these properties you can make a call to your database, and return the proper language key. Alternatively if you want some kind of fallback to a default language you can also specify that here.
```

Also add at the bottom of the document:
```
[`saphclientoptions`]: ../../Documentation/api-framework/interfaces/SapphireClientOptions
[`fetchLanguage`]: ../../Documentation/api-plugins/classes/sapphire_plugin_i18next.InternationalizationHandler#fetchlanguage
[`InternationalizationContext`]: ../../Documentation/api-plugins/interfaces/sapphire_plugin_i18next.InternationalizationContext
```

```typescript ts2esm2cjs|{3}|{3}
import { SapphireClient } from '@sapphire/framework';
import { InternationalizationContext } from '@sapphire/plugin-i18next';

const client = new SapphireClient({
  intents: ['GUILDS', 'GUILD_MESSAGES'],
  i18n: {

>     fetchLanguage: async (context: InternationalizationContext) => {
>       if (!context.guild) return process.env.DEFAULT_LOCALE;
>       const guildSettings = await db.find({ guild_id: context.guildId });
>       return guildSettings.locale;
>     }

Reviewer: ```suggestion
    fetchLanguage: async (context: InternationalizationContext) => {
      if (!context.guild) return 'en-US'; // Or whatever your default language is
      const guildSettings = await db.find({ guild_id: context.guildId }); // example of querying your database. The exact syntax will depend on your ORM
      return guildSettings.language;
    }
```

</TabItem>
</Tabs>

## Get the language set per server

To obtain the language that each server has configured, as well as the custom prefixes for each server, for that we will
have now an option called `i18n` in the Sapphire client options, in which we will do the following.

```typescript ts2esm2cjs|{3}|{3}
import { SapphireClient } from '@sapphire/framework';
import { InternationalizationContext } from '@sapphire/plugin-i18next';

> import '@sapphire/plugin-i18next/register';

Reviewer: ```suggestion
```

register breaks the convertion so it needs its own codeblock which is already above.

able to configure their own language through some kind of configuration command while you store this data in a database.
While this page will not cover those aspects, what we will cover here is how you can inform the i18next plugin about the
language that should be used in the current context.

In order to achieve this, you will need to provide the options to the plugin. This is done through the `i18next` option
on the [`ClientOptions`][saphclientoptions], specifically the [`fetchLanguage`] property on that object. This method
receives 1 parameter, which is an object that has `message`, `guild`, and `user`, we call it the
[`InternationalizationContext`]. Using these properties you can make a call to your database, and return the proper
language key. Alternatively if you want some kind of fallback to a default language you can also specify that here.

```typescript ts2esm2cjs|{3}|{3}
import { SapphireClient } from '@sapphire/framework';
import { InternationalizationContext } from '@sapphire/plugin-i18next';

const client = new SapphireClient({
  intents: ['GUILDS', 'GUILD_MESSAGES'],
  i18n: {
    fetchLanguage: async (context: InternationalizationContext) => {

>       // Of whatever your default language is

Reviewer: ```suggestion
      // Or whatever your default language is
```

Woops typo (my bad I think)
